src/main/kotlin/cacophony/pipeline/CacophonyPipeline.kt

```diff
@@ -268,8 +269,10 @@ class CacophonyPipeline(
                             instructionCovering,
                             functionHandlers[functionDeclaration]!!,
                             loweredCfg,
+                            liveness[functionDeclaration]!!,
```

Comment:
probably not a thing to do right now, but a potential refactor in the future - pipeline doesn't seem like the right place for this code to live. What are your thoughts?

Reply:
i agree, I hate the current state of pipeline

Reply:
but also I agree that this is a task for future refactor

Reply:
The thing is that spill handling has so many dependencies that it does not fit the "pipeline" paradigm, I'm not sure how this could look better though, perhaps "spill handling" should be a part of internal logic of register allocation